### PR TITLE
Merge fluture-template

### LINKED ISSUE #13: Make the tests run on Node 12
See #12

### PR DESCRIPTION
**Pull request by dicearr**
Although this is not a Fluture related project I think it would benefit from using `fluture-template`.

This  aims to finally resolve #15 and close #13 and close #12.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -191,11 +171,11 @@ State['fantasy-land/chainRec'] = function chainRec(f, v) {
 //. 2
 //. ```
 State.prototype['fantasy-land/chain'] = function chain(f) {
-  var self = this;
-  return new State (function(s) {
-    var r = self.run (s);
+  const self = this;
+  return new State ((s => {
+    const r = self.run (s);
```
Comment (Avaq) [inline index.js:176]:
```suggestion
  return new State ((s => {
    const r = this.run (s);
```

File: index.js [hunk 2]
```diff
@@ -465,11 +420,11 @@ export function StateT(M) {
   //. Z.of (Maybe, 2)
   //. ```
   StateT.prototype['fantasy-land/ap'] = function(mf) {
-    var mx = this;
-    return new StateT (function(state) {
-      var get = StateT.evalState (state); // Monad {st, val}
+    const mx = this;
```
Comment (Avaq) [inline index.js:423]: This is no longer strictly necessary, although, since it also clarifies the role of `this`, you may want to keep it.

Comment (dicearr) [reply index.js:423]: I think I am going to keep it.

File: index.js [hunk 3]
```diff
@@ -429,13 +385,11 @@ export function StateT(M) {
...
-      return Z.chain (
-        function(state) { return f (state.value).run (state.state); },
-        self.run (s)
-      );
-    });
+    const self = this;
+    return new StateT (s => Z.chain (
+      state => f (state.value).run (state.state),
+      self.run (s)
+    ));
```
Comment (Avaq) [inline index.js:392]:
```suggestion
    return new StateT (s => Z.chain (
      state => f (state.value).run (state.state),
      this.run (s)
    ));
```

What about this one?

Comment (dicearr) [reply index.js:392]: Didn't see that one. I will review all of them this time :)

Comment (Avaq) [reply index.js:392]: Somehow I also missed this one last time.

**Pull request merged**